processing/src/main/java/io/druid/query/topn/TopNQueryEngine.java

```
@@ -124,4 +125,13 @@ public TopNQueryEngine(StupidPool<ByteBuffer> bufferPool)
 
     return new TopNMapFn(query, topNAlgorithm);
   }
+
+  public static boolean canApplyExtractionInPost(TopNQuery query)
+  {
+    return query != null
```

So throw an error instead of silently return false?